### ISSUE TITLE
Seem to reliably lose last ping under certain parameters when average latency > 0.5 * host interval

### ISSUE THREAD

**Issue by initialed85** [schweikert/fping#120]
For the following latency settings-

`$ qdisc change dev ens160 root netem delay 0ms`
`$ qdisc change dev ens160 root netem delay 500ms`
`$ qdisc change dev ens160 root netem delay 501ms`
`$ qdisc change dev ens160 root netem delay 1000ms`
`$ qdisc change dev ens160 root netem delay 1500ms`
`$ qdisc change dev ens160 root netem delay 1501ms`
`$ qdisc change dev ens160 root netem delay 2000ms`

I get the following results-

`# fping -b 12 -B 1.0 -c 8 -i 1 -p 1000 -q 10.10.0.137`
`10.10.0.137 : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.29/0.38/0.51`

`# fping -b 12 -B 1.0 -c 8 -i 1 -p 1000 -q 10.10.0.137`
`10.10.0.137 : xmt/rcv/%loss = 8/8/0%, min/avg/max = 500/500/500`

`# fping -b 12 -B 1.0 -c 8 -i 1 -p 1000 -q 10.10.0.137`
`10.10.0.137 : xmt/rcv/%loss = 8/7/12%, min/avg/max = 501/501/501`

`# fping -b 12 -B 1.0 -c 8 -i 1 -p 1000 -q 10.10.0.137`
`10.10.0.137 : xmt/rcv/%loss = 8/7/12%, min/avg/max = 1000/1000/1000`

`# fping -b 12 -B 1.0 -c 8 -i 1 -p 1000 -q 10.10.0.137`
`10.10.0.137 : xmt/rcv/%loss = 8/7/12%, min/avg/max = 1500/1500/1500`

`# fping -b 12 -B 1.0 -c 8 -i 1 -p 1000 -q 10.10.0.137`
`10.10.0.137 : xmt/rcv/%loss = 8/6/25%, min/avg/max = 1501/1501/1501`

`# fping -b 12 -B 1.0 -c 8 -i 1 -p 1000 -q 10.10.0.137`
`10.10.0.137 : xmt/rcv/%loss = 8/6/25%, min/avg/max = 2000/2000/2000`

As you can see, fping is returning that it failed to receive all the pings (probably due to some hard cutoff as a result of count * host interval).

I would have expected that up to maximum duty cycle for this configuration (1000ms), I should have been able to maintain 8/8 pings received.

In practice, the tipping point seems to be 501ms for a ping to be regarded as lost.

**Comment by schweikert** [comment]
What version of fping are you using? This would be expected with fping < 4.0, but not with fping >= 4.0.

**Comment by initialed85** [comment]
Hi David, 

Thanks for following up- we use the version available in the Ubuntu 16.04 package repo which seems to be:

    # fping -v
    fping: Version 3.13
    fping: comments to david@schweikert.ch

So I guess that would explain it! I'll change our Dockerfile to build the latest fping from source.

Thanks again,

Ed
